### PR TITLE
Ops 3940 create up to 5 sqs for a microservice

### PR DESCRIPTION
**Pull request by ronny-panknin-flaconi**
### tickets with dependency:

https://flaconi.atlassian.net/browse/ALPHA-2398
https://flaconi.atlassian.net/browse/ALPHA-2485
https://flaconi.atlassian.net/browse/ALPHA-2485

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -334,3 +334,93 @@ output "this_aws_s3_bucket_arn" {
     0,
   )
 }
+
+# -------------------------------------------------------------------------------------------------
+# SQS 1
+# -------------------------------------------------------------------------------------------------
+output "sqs_queue_name" {
```
Comment (cytopia) [inline outputs.tf:341]:
```suggestion
output "sqs1_queue_name" {
```

File: outputs.tf [hunk 2]
```diff
@@ -334,3 +334,93 @@ output "this_aws_s3_bucket_arn" {
...
+  description = "SQS queue name"
+  value       = module.sqs1.this_sqs_queue_name
+}
+
+output "sqs_queue_id" {
```
Comment (cytopia) [inline outputs.tf:346]:
```suggestion
output "sqs1_queue_id" {
```

File: outputs.tf [hunk 3]
```diff
@@ -334,3 +334,93 @@ output "this_aws_s3_bucket_arn" {
...
+  description = "SQS queue ID"
+  value       = module.sqs1.this_sqs_queue_id
+}
+
+output "sqs_queue_arn" {
```
Comment (cytopia) [inline outputs.tf:351]:
```suggestion
output "sqs1_queue_arn" {
```

File: README.md [hunk 4]
```diff
@@ -411,6 +451,21 @@ The following resources _CAN_ be created:
...
+| sqs3\_queue\_arn | SQS queue ARN |
+| sqs3\_queue\_id | SQS queue ID |
+| sqs3\_queue\_name | SQS queue name |
+| sqs4\_queue\_arn | SQS queue ARN |
+| sqs4\_queue\_id | SQS queue ID |
+| sqs4\_queue\_name | SQS queue name |
+| sqs5\_queue\_arn | SQS queue ARN |
+| sqs5\_queue\_id | SQS queue ID |
+| sqs5\_queue\_name | SQS queue name |
+| sqs\_queue\_arn | SQS queue ARN |
```
Comment (cytopia) [inline README.md:466]:
```suggestion
| sqs1\_queue\_arn | SQS queue ARN |
```

File: README.md [hunk 5]
```diff
@@ -411,6 +451,21 @@ The following resources _CAN_ be created:
...
+| sqs\_queue\_id | SQS queue ID |
```
Comment (cytopia) [inline README.md:467]:
```suggestion
| sqs1\_queue\_id | SQS queue ID |
```

File: README.md [hunk 6]
```diff
@@ -411,6 +451,21 @@ The following resources _CAN_ be created:
...
+| sqs\_queue\_name | SQS queue name |
```
Comment (cytopia) [inline README.md:468]:
```suggestion
| sqs1\_queue\_name | SQS queue name |
```

File: sqs.tf [hunk 7]
```diff
@@ -0,0 +1,18 @@
+locals {
```
Comment (cytopia) [inline sqs.tf:1]: I would also rename this file to `sqs1.tf`

**Review by cytopia — DISMISSED** [review]
For consistency

File: sqs.tf (hunk 7 above)
Comment (ronny-panknin-flaconi) [reply sqs.tf:1]: ok

File: sqs2.tf [hunk 8]
```diff
@@ -0,0 +1,18 @@
+locals {
+  sqs2_name = length(var.sqs2_name_override) > 0 ? var.sqs2_name_override : join("-", [var.name, "1"])
```
Comment (cytopia) [inline sqs2.tf:2]: Shouldn't it be like so (`2` at the very end)?
```suggestion
  sqs2_name = length(var.sqs2_name_override) > 0 ? var.sqs2_name_override : join("-", [var.name, "2"])
```

File: sqs3.tf [hunk 9]
```diff
@@ -0,0 +1,18 @@
+locals {
+  sqs3_name = length(var.sqs3_name_override) > 0 ? var.sqs3_name_override : join("-", [var.name, "1"])
```
Comment (cytopia) [inline sqs3.tf:2]: Shouldn't it be like so (`3` at the very end)?
```suggestion
  sqs3_name = length(var.sqs3_name_override) > 0 ? var.sqs3_name_override : join("-", [var.name, "3"])
```

File: sqs4.tf [hunk 10]
```diff
@@ -0,0 +1,18 @@
+locals {
+  sqs4_name = length(var.sqs4_name_override) > 0 ? var.sqs4_name_override : join("-", [var.name, "1"])
```
Comment (cytopia) [inline sqs4.tf:2]: Shouldn't it be like so (`4` at the very end)?
```suggestion
  sqs4_name = length(var.sqs4_name_override) > 0 ? var.sqs4_name_override : join("-", [var.name, "4"])
```

File: sqs5.tf [hunk 11]
```diff
@@ -0,0 +1,18 @@
+locals {
+  sqs5_name = length(var.sqs5_name_override) > 0 ? var.sqs5_name_override : join("-", [var.name, "1"])
```
Comment (cytopia) [inline sqs5.tf:2]: Shouldn't it be like so (`5` at the very end)?
```suggestion
  sqs5_name = length(var.sqs5_name_override) > 0 ? var.sqs5_name_override : join("-", [var.name, "5"])
```

Comment (ronny-panknin-flaconi) [reply sqs5.tf:2]: mr eagle eye joined the game?
you´re correct but we also could keep it to force the users to create custom names for the sqs, in the other case they will delete the sqs after creating with  1,1,1,1,1 :D

**Review by cytopia — APPROVED** [review]
Nice work :+1:

**Review by danvaida — COMMENTED** [review]
Is it not possible to simply have a variable like an integer to use in a loop and create that many resources?
Or even a list of maps, if one needs more characteristics.
This is going against the DRY principle so badly...

**Pull request merged**